src/encrypt.rs

```diff
@@ -0,0 +1,107 @@
+pub(crate) struct EncryptionConfiguration {
+    pub(crate) kind: String,
+    pub(crate) apiVersion: String,
+    pub(crate) resources: Vec<Resource>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub(crate) struct Resource {
+    pub(crate) resources: Vec<String>,
+    pub(crate) providers: Vec<Provider>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub(crate) struct Provider {
+    pub(crate) aesgcm: Option<AesGcm>,
+    pub(crate) aescbc: Option<AesCbc>,
+    pub(crate) identity: Option<Identity>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub(crate) struct AesGcm {
+    pub(crate) keys: Vec<Key>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub(crate) struct AesCbc {
+    pub(crate) keys: Vec<Key>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub(crate) struct Key {
+    pub(crate) name: String,
+    pub(crate) secret: String,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub(crate) struct Identity {}
+
+#[derive(Clone)]
+pub(crate) struct ResourceTransformers {
+    pub(crate) resource_to_prefix_transformers: HashMap<String, Vec<Box<dyn Transformer + Send + Sync>>>,
+}
+
+impl EncryptionConfiguration {
+    pub(crate) async fn parse_from_file(config_bytes: &[u8]) -> Result<Self> {
+        let config: EncryptionConfiguration = serde_json::from_slice(config_bytes)?;
+
+        ensure!(config.kind == KIND, format!("kind should equal {}", KIND));
+        ensure!(config.apiVersion == API_VERSION, format!("apiVersion should equal {}", API_VERSION));
+
+        Ok(config)
+    }
+}
+
+impl ResourceTransformers {
+    #[allow(clippy::if_same_then_else)]
+    pub(crate) async fn parse_from_encryption_configuration(config: EncryptionConfiguration) -> Result<Self> {
+        let mut transformers: HashMap<String, Vec<Box<dyn Transformer + Send + Sync>>> = HashMap::new();
+
+        for resources in config.resources {
+            let mut providers: Vec<Box<dyn Transformer + Send + Sync>> = Vec::new();
+
+            for provider in resources.providers {
+                if provider.aesgcm.is_some() {
+                    let prefix = "k8s:enc:aesgcm:v1:";
+                    for key in provider.aesgcm.context("no AES-GCM provider")?.keys {
+                        let prefix = format!("{}{}:", prefix, key.name);
+                        providers.push(Box::new(aesgcm::AesGcm::new(prefix, key.secret)));
+                    }
+                } else if provider.aescbc.is_some() {
+                    let prefix = "k8s:enc:aescbc:v1:";
+                    for key in provider.aescbc.context("no AES-CBC provider")?.keys {
+                        let prefix = format!("{}{}:", prefix, key.name);
+                        providers.push(Box::new(aescbc::AesCbc::new(prefix, key.secret)));
+                    }
+                } else if provider.identity.is_some() {
+                    // Nothing to implement
+                } else {
+                    bail!("unsupported provider");
+                }
```

Comment:
`match` patterns are a good fit here:

```suggestion
                match (provider.aesgcm, provider.aescbc, provider.identity) {
                    (Some(aesgcm), None, None) => {
                        let prefix = "k8s:enc:aesgcm:v1:";
                        for key in aesgcm.keys {
                            let prefix = format!("{}{}:", prefix, key.name);
                            providers.push(Box::new(aesgcm::AesGcm::new(prefix, key.secret)));
                        }
                    }
                    (None, Some(aescbc), None) => {
                        let prefix = "k8s:enc:aescbc:v1:";
                        for key in aescbc.keys {
                            let prefix = format!("{}{}:", prefix, key.name);
                            providers.push(Box::new(aescbc::AesCbc::new(prefix, key.secret)));
                        }
                    }
                    (None, None, Some(_)) => {
                        // Nothing to implement
                    }
                    _ => {
                        bail!("unsupported provider");
                    }
                }

```